logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +190,22 @@ def configure_plugin_paths(paths)
     end
   end
 
+  def agent_class
+    return @agent_class if @agent_class # We don't want to load these things twice
+
+    LogStash::AgentPluginRegistry.load_all
```

Comment:
since we're loading all anyway I don't see why it needs to be hidden under the agent_class getter, I'd suggest making this part of the runner initialize or something similar..this way we can be clearer during the runner#execute:

```ruby
@agent = LogStash::AgentPluginRegistry.lookup(agent_name).new
```